src/note.rs

```diff
@@ -237,8 +237,8 @@ impl Note {
     }
 
     /// Return the position of the note on the tree.
-    pub fn pos(&self) -> u64 {
-        self.pos
+    pub fn pos(&self) -> &u64 {
```

Comment:
Since you modify in message and crossover the methods to be `const` whenever possible, you might do the same for note as well.

Reply:
Fixed